lib/src/geometry/collision_detection.dart

```diff
@@ -1,18 +1,32 @@
 import '../../extensions.dart';
 import '../components/mixins/collidable.dart';
 
+void aloneCollisionDetection(Collidable collidableX, Collidable collidableY) {
```

Comment:
This needs a clearer name for what it does.
I'll try to come up with a suggestion.